bluetooth_mesh/application.py

```diff
@@ -534,7 +534,7 @@ def _unregister(self):
         self.elements = {}
 
     async def connect(
-        self, addr, iv_index=0
+        self, addr: Optional[int] = None, iv_index: int = 0, addr_getter: Optional[Awaitable[int]] = None
```

Comment:
I'd rather have this overloaded, so that `addr` can be either an `int`, or `Awaitable[int]` (or maybe even `Callable[int]` for regular callbacks), and then use `inspect` below.